pkg/disttask/framework/proto/task.go

```diff
@@ -146,16 +148,19 @@ func (t *TaskBase) CompareTask(other *Task) int {
 // Compare compares two tasks by task rank.
 // returns < 0 represents rank of t is higher than 'other'.
 func (t *TaskBase) Compare(other *TaskBase) int {
-	if t.Priority != other.Priority {
-		return t.Priority - other.Priority
+	if r := cmp.Compare(t.Priority, other.Priority); r != 0 {
+		return r
 	}
-	if t.CreateTime != other.CreateTime {
-		if t.CreateTime.Before(other.CreateTime) {
-			return -1
-		}
-		return 1
+	if r := t.CreateTime.Compare(other.CreateTime); r != 0 {
+		return r
 	}
-	return int(t.ID - other.ID)
+	return cmp.Compare(t.ID, other.ID)
+}
+
+// String implements fmt.Stringer interface.
+func (t *TaskBase) String() string {
```

Comment:
if we use JSON, we don't need to adjust the code if we add more fields in future.

Reply:
just found I ever write in this way here 😓
https://github.com/pingcap/tidb/blob/4407ef5c2253ea56b838021b954def4bd83c4dca/pkg/executor/importer/job.go#L94-L98
both ways lgtm, currently, I slightly prefer avoiding error here, including unhandled